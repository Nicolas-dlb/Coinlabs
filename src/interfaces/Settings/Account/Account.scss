.settings_account {
  transition: all 0.3s ease;
  width: 85%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);

  .user_card {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    height: 13vh;
    background-color: var(--third-bg-color);
    border-radius: 10px;
    width: 100%;
    transition: background-color 0.3s ease;

    .user_card_img {
      width: 80px;
      position: relative;
      top: 1vh;
      img {
        width: 80px;
        height: 80px;
        border-radius: 25%;
      }
      .point {
        width: 10px;
        height: 10px;
        background: #8fbd42;
        border-radius: 50%;
        position: relative;
        top: -1.9vh;
        left: 70px;
      }
    }
    .user_card_info {
      display: flex;
      flex-direction: column;
      justify-content: space-evenly;
      align-items: flex-start;
      height: 90%;
      margin-left: 22px;
      width: fit-content;
      h5 {
        color: var(--secondary-font-color);
      }
    }
    .change_user {
      display: flex;
      color: var(--third-font-color);
      font-size: 15px;
      .change {
        padding-left: 5px;
        cursor: pointer;
        position: relative;
        top: 1px;
        font-size: 14px;
        color: var(--primary-font-color);
        text-decoration: underline;
        &:hover {
          color: rgb(233, 223, 223);
        }
      }
    }
  }
}

@media only screen and (min-width: 992px) {
  .settings_account {
    width: 320px;
    padding: 13px 20px;
    .account {
      margin-left: 0vw;
    }
    .point {
      left: 92% !important;
    }
  }
}
