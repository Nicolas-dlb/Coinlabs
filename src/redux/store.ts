import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
