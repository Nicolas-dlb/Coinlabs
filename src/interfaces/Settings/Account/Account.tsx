import "./Account.scss";
import React from "react";
import {
  selectProfilPic,
  selectUserEmail,
  selectUserName,
} from "redux/reducers/userSlice";
import { useSelector } from "react-redux";
import { auth } from "firebaseConfig";

function Account() {
  const profilPic = useSelector(selectProfilPic);
  const username = useSelector(selectUserName);
  const email = useSelector(selectUserEmail);
  return (
    <div className="settings_account">
      <h3 className="account">Account</h3>
      <div className="user_card">
        <div className="user_card_img">
          <img
            src={
              profilPic ||
              "https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
            }
            alt=""
          />
          <div className="point" />
        </div>
        <div className="user_card_info">
          <h3>{username}</h3>
          <h5>{email}</h5>
          <div className="change_user">
            <p>Not you?</p>
            <span
              className="change"
              tabIndex={0}
              role="button"
              onKeyDown={() => {
                auth.signOut();
              }}
              onClick={() => {
                auth.signOut();
              }}
            >
              Change User
            </span>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Account;
