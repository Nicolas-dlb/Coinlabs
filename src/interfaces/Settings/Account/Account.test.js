import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import store from "redux/store";
import { Provider } from "react-redux";
import Account from "./Account";

describe("Account", () => {
  let AccountComponent;
  beforeEach(async () => {
    await act(async () => {
      AccountComponent = mount(
        <Provider store={store}>
          <Account />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(AccountComponent).toMatchSnapshot();
  });
  it("should render Account title", () => {
    expect(AccountComponent.find("h3").first().text()).toBe("Account");
  });
  it("should render user Profil picture", () => {
    expect(AccountComponent.find("img").first().exists()).toBe(true);
  });
  it("should render user name", () => {
    expect(AccountComponent.find("h3").first().exists()).toBe(true);
  });
  it("should render user email", () => {
    expect(AccountComponent.find("h5").exists()).toBe(true);
  });
  it("should render a logout", () => {
    expect(AccountComponent.find(".change_user").first().exists()).toBe(true);
  });
});
