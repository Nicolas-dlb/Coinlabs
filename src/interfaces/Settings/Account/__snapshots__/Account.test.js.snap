// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Account>
    <div
      className="settings_account"
    >
      <h3
        className="account"
      >
        Account
      </h3>
      <div
        className="user_card"
      >
        <div
          className="user_card_img"
        >
          <img
            alt=""
            src="https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
          />
          <div
            className="point"
          />
        </div>
        <div
          className="user_card_info"
        >
          <h3 />
          <h5 />
          <div
            className="change_user"
          >
            <p>
              Not you?
            </p>
            <span
              className="change"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Change User
            </span>
          </div>
        </div>
      </div>
    </div>
  </Account>
</Provider>
`;
